File: iOS/sampleObjectC/sampleObjectC/MessageEventCollectionViewCell.m
//
//  MessageEventCollectionViewCell.m
//  sampleObjectC
//
//  Created by oshumini on 2017/6/16.
//  Copyright © 2017年 HXHG. All rights reserved.
//

#import "MessageEventCollectionViewCell.h"

@interface MessageEventCollectionViewCell ()
@property(strong, nonatomic)UILabel *evenText;
@end

@implementation MessageEventCollectionViewCell

- (instancetype)initWithFrame:(CGRect)frame
{
  self = [super initWithFrame:frame];
  if (self) {
    _evenText = [[UILabel alloc] initWithFrame: CGRectZero];
    [self.contentView addSubview: _evenText];
    _evenText.textColor = [[UIColor alloc] initWithNetHex: 0x7587A8];
    _evenText.textAlignment = NSTextAlignmentCenter;
    
  }
  return self;
}

- (void)presentCell:(NSString *)eventText {
  _evenText.text = eventText;
  _evenText.frame = CGRectMake(0, 0, 300, 20);
  
  _evenText.center = CGPointMake(self.contentView.frame.size.width/2,
                                 self.contentView.frame.size.height/2);
  
}

@end

File: iOS/sample/sample/MessageEventCollectionViewCell.swift
//
//  MessageEventCollectionViewCell.swift
//  sample
//
//  Created by oshumini on 2017/6/16.
//  Copyright © 2017年 HXHG. All rights reserved.
//

import UIKit

class MessageEventCollectionViewCell: UICollectionViewCell {
  var eventLabel = UILabel()
  
  override init(frame: CGRect) {
    super.init(frame: frame)
    
    self.contentView.addSubview(eventLabel)
    eventLabel.frame = CGRect(x: 0, y: 0, width: 300, height: 20)
    eventLabel.textColor = UIColor(netHex: 0x7587A8)
    eventLabel.textAlignment = .center
  }
  
  func presentCell(eventText: String) {
    eventLabel.text = eventText
    eventLabel.center = CGPoint(x: self.contentView.imui_width/2, y: self.contentView.imui_height/2)
  }
  
  required init?(coder aDecoder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }
}
